app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/TextInputViewModel.kt

```diff
@@ -28,20 +30,47 @@ class TextInputViewModel private constructor(
 ) : StateItemViewModel(ViewType.TEXT_INPUT_INTERACTION), InteractionAnswerHandler {
   var answerText: CharSequence = ""
   val hintText: CharSequence = deriveHintText(interaction)
+  private var pendingAnswerError: String? = null
 
   var isAnswerAvailable = ObservableField<Boolean>(false)
+  val errorMessage = ObservableField<String>("")
 
   init {
     val callback: Observable.OnPropertyChangedCallback =
       object : Observable.OnPropertyChangedCallback() {
         override fun onPropertyChanged(sender: Observable, propertyId: Int) {
           interactionAnswerErrorOrAvailabilityCheckReceiver.onPendingAnswerErrorOrAvailabilityCheck(
-            /* pendingAnswerError= */ null,
-            answerText.isNotEmpty()
+            /* pendingAnswerError= */ pendingAnswerError,
```

Comment:
```suggestion
            pendingAnswerError = pendingAnswerError,
```

For consistency, and this is more idiomatic in Kotlin.

Reply:
Okay.